src/hooks/useGrantContract.ts

```diff
@@ -0,0 +1,70 @@
+import { ethers } from 'ethers';
+import { useMemo, useState } from 'react';
+import { TokenVestingController__factory } from 'contracts/types';
+
+import { useWeb3React } from '@web3-react/core';
+
+import { GrantContract, Maybe } from '../util';
+
+enum errorState {
+	NoError = 'Checking...',
+	NotOwner = 'You are not the Owner of this address.',
+	NotChainId = 'No chain id detected.',
+	NotSuportChainId = 'Chain id not supported.',
+	NoAddress = 'Invalid vesting contract address.',
+}
+
+export function useGrantContract(address: string) {
+	const context = useWeb3React<ethers.providers.Web3Provider>();
+
+	const [isError, setError] = useState('');
+	const [owner, setOwner] = useState('');
+
+	const { account, library, chainId } = context;
+	const contract = useMemo((): Maybe<GrantContract> => {
```

Comment:
This is a confusing return type;

`GrantContract` is an interface which contains one member `grantableVesting` which is a `TokenVestingController`.

This memo should instead return a `Maybe<TokenVestingController>` since this wrapper doesn't do anything and makes the code harder to understand